test/integration/lib/make_suite.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+const Config = require('config-js');
+
+var config = new Config('./test/integration/lib/config/foxbox.js');
+var philipshue_server = require('../lib/philipsHue_server.js');
+var ipcamera_server = require('../lib/ipcamera_server.js');
+var webPush_server = require('../lib/webpush_server.js');
+var nupnp_server = require('../lib/nupnp_PhilipsHue.js');
+var foxboxManager = require('../lib/foxbox_process_manager.js');
+var config = new Config('./test/integration/lib/config/foxbox.js');
+
+
+var make_suite = (function() {
+  var hueSimOn = false;
+  var cameraSimOn = false;
+  var webPushSimOn = false;
+  
+  function turnAllSims() { 
+      
+    //Start the foxbox and associated simulators
+    turnOnHue();
+    turnOnCamera(); 
+    turnOnWebPush();
+ }
+
+  function turnOnHue(){
+    nupnp_server.start(config.get('nupnp_server.id'),
+      config.get('philips_hue.url') +':'+config.get('philips_hue.port'),
+      config.get('nupnp_server.port'));
+
+    philipshue_server.setup(config.get('philips_hue.port'));
+    hueSimOn = true;
+  }
+
+  function turnOnCamera() {
+    ipcamera_server.setup();
+    cameraSimOn = true;
+  }
+
+  function turnOnWebPush() {
+    webPush_server.setup();
+    webPushSimOn = true;
+  }
+
+  function turnOnFoxbox(done) {
+    foxboxManager.fullOptionStart(done);
+  }
+
+  function nupnpOff() {
+    nupnp_server.stop();
+  }
+
+  function hueOff() {
+    hueSimOn = false;
+    return philipshue_server.stop();
+  }
+
+  function cameraOff() {
+    cameraSimOn = false;
+    return ipcamera_server.stop();
+  }
+
+  function webPushOff() {
+    webPushSimOn = false;
+    return webPush_server.stop();
+  }
+
+  function makeSuite(desc, test) {
+    describe(desc, function () {
+      this.timeout(500000);
```

Comment:
This timeout won't pass on Travis. There's a 10-minute-limit there. I'd suggest to downgrade the time to 30 or 60 seconds.